From 21eeb6ec8e4c9a53a5509afe874b9b7268514a13 Mon Sep 17 00:00:00 2001
From: Muhammad Saheed <saheed@FreeBSD.org>
Date: Thu, 10 Jul 2025 18:18:30 +0530
Subject: [PATCH] sysutils/wutil: Add new port

wutil provides a CLI and TUI for WiFi management. It interacts with
wpa_supplicant over its control interface.

Sponsored by:	Google LLC (GSoC)
---
 sysutils/wutil/Makefile  | 32 ++++++++++++++++++++++++++++++++
 sysutils/wutil/distinfo  |  3 +++
 sysutils/wutil/pkg-descr |  2 ++
 3 files changed, 37 insertions(+)
 create mode 100644 sysutils/wutil/Makefile
 create mode 100644 sysutils/wutil/distinfo
 create mode 100644 sysutils/wutil/pkg-descr

diff --git a/sysutils/wutil/Makefile b/sysutils/wutil/Makefile
new file mode 100644
index 0000000000..b989865807
--- /dev/null
+++ b/sysutils/wutil/Makefile
@@ -0,0 +1,32 @@
+PORTNAME=	wutil
+DISTVERSIONPREFIX=	v
+DISTVERSION=	0.2.0
+CATEGORIES=	sysutils
+
+MAINTAINER=	saheed@FreeBSD.org
+COMMENT=	CLI and TUI for WiFi Management
+WWW=		https://github.com/MainKt/wutil
+
+LICENSE=	BSD2CLAUSE
+LICENSE_FILE=	${WRKSRC}/LICENSE
+
+LIB_DEPENDS+=	libifconfig.so:net/libifconfig
+LIB_DEPENDS+=	libwpa_client.so:security/wpa_supplicant
+
+MAKE_ARGS+=	CFLAGS="-I${LOCALBASE}/include" LDFLAGS="-L${LOCALBASE}/lib"
+
+USE_GITHUB=	yes
+GH_ACCOUNT=	MainKt
+
+PLIST_FILES=	bin/wutil \
+							bin/wutui \
+							share/man/man8/wutil.8.gz \
+							share/man/man8/wutui.8.gz
+
+do-install:
+	${INSTALL_PROGRAM} ${WRKSRC}/wutil ${STAGEDIR}${PREFIX}/bin
+	${INSTALL_PROGRAM} ${WRKSRC}/wutui ${STAGEDIR}${PREFIX}/bin
+	${INSTALL_MAN} ${WRKSRC}/wutil.8 ${STAGEDIR}${PREFIX}/share/man/man8
+	${INSTALL_MAN} ${WRKSRC}/wutui.8 ${STAGEDIR}${PREFIX}/share/man/man8
+
+.include <bsd.port.mk>
diff --git a/sysutils/wutil/distinfo b/sysutils/wutil/distinfo
new file mode 100644
index 0000000000..e7b56d746e
--- /dev/null
+++ b/sysutils/wutil/distinfo
@@ -0,0 +1,3 @@
+TIMESTAMP = 1756052491
+SHA256 (MainKt-wutil-v0.2.0_GH0.tar.gz) = 449b4292405463ba387c7c297804c7ab204c6d1a607aa6b298f521b771ffb95e
+SIZE (MainKt-wutil-v0.2.0_GH0.tar.gz) = 23758
diff --git a/sysutils/wutil/pkg-descr b/sysutils/wutil/pkg-descr
new file mode 100644
index 0000000000..42cdebf0d4
--- /dev/null
+++ b/sysutils/wutil/pkg-descr
@@ -0,0 +1,2 @@
+wutil is a WiFi management utility for FreeBSD.
+It provides a CLI, wutil, and a TUI, wutui.
-- 
2.49.0

